nocasedict/_nocasedict.py

```diff
@@ -576,42 +625,45 @@ def update(self, *args, **kwargs):
         """
         Update the dictionary from key/value pairs.
 
-        If an item for a key exists in the dictionary (looked up
-        case-insensitively), its value is updated. If an item for a key does
-        not exist, it is added to the dictionary.
+        If a key is already present in the dictionary (looked up
+        case-insensitively), its key and value is updated (without affecting
+        its position in the dictionary iteration order). Otherwise, an item
+        with the key and value is added to the dictionary.
 
         The provided key and value objects will be referenced from the
-        dictionary without being copied.
+        dictionary without being copied, consistent with the built-in
+        :class:`py:dict` class.
 
         Parameters:
 
-          *args : An optional single positional argument that must be one of:
+          *args : An optional single positional argument representing key-value
+            pairs to update the dictionary from, in iteration order of the
+            specified object. The argument must be one of:
 
-            - a dictionary object, or more generally an object that supports
-              subscription by key and has a method ``keys()`` returning an
-              iterable of keys.
+            - a dictionary object, or more specifically an object that has a
+              method ``keys()`` providing iteration through the keys and that
+              supports subscription by key for accessing the values.
```

Comment:
what is the meaning of "subscription by key"?

Reply:
Subscription means using `x[]`, subscription by key means `x[key]` as opposed to subscription by index `x[42]`.